sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1308,6 +1312,16 @@ object CheckCartesianProducts extends Rule[LogicalPlan] with PredicateHelper {
     if (SQLConf.get.crossJoinEnabled) {
       plan
     } else plan transform {
+      case j @ Join(_, _, _, condition)
+          if condition.isDefined && PullOutPythonUDFInJoinCondition.hasPythonUDF(condition.get) =>
+        // if the crossJoinEnabled is false, a RuntimeException will be thrown later while
+        // the PythonUDF need to access both side of join, we throw firstly here for better
+        // readable information.
+        throw new AnalysisException(s"Detected the join condition:${j.condition} of this join " +
```

Comment:
when will we hit it? If the python udf can't be pulled out because of join type problems, we already throw exception at `PullOutPythonUDFInJoinCondition`. Did I miss something?